Sources/XCHammer/XcodeTarget.swift

```diff
@@ -432,6 +432,29 @@ public class XcodeTarget: Hashable, Equatable {
                     // either way, get the dependencies
                     return xcodeTarget.xcDependencies
                 }
+
+                // Do not link static libraries that aren't going to exist.
+                // These targets still need to be included in the project.
+                if case xcodeTarget.extractProductType() = ProductType.StaticLibrary  {
+                    let compiledSrcs = xcodeTarget.sourceFiles.filter {
+                       fileInfo in
+                       let path = fileInfo.subPath
+                       if path.hasSuffix(".swift")
+                            || path.hasSuffix(".m")
+                            || path.hasSuffix(".mm")
+                            || path.hasSuffix(".cpp")
+                            || path.hasSuffix(".c")
+                            || path.hasSuffix(".cc")
+                            || path.hasSuffix(".S") {
```

Comment:
what about 
```guard let suffix = components(separatedBy:).last { return false}
return [<static list of suffixes>].contains(suffix)
```

Reply:
OR alternatively
```
// as an enum somewhere
enum JerrysSourceTypes: String {
    case swift = "swift"
    case m = "m"
    case mm = "mm"
    case cpp = "cpp"
    case c = "c"
    case cc = "cc"
    case S = "S"
}

guard let suffix = path.components(separatedBy: ".").last { return false}
return JerrysSourceTypes(rawValue: suffix) != nil
```

Reply:
I like the idea of putting known extension types into a `Set` which would give fast lookup 👍